clients/src/main/java/org/apache/kafka/clients/admin/internals/AdminApiDriver.java

```diff
@@ -260,12 +261,18 @@ public void onFailure(
                 .filter(future.lookupKeys()::contains)
                 .collect(Collectors.toSet());
             retryLookup(keysToUnmap);
+        } else if (t instanceof UnsupportedVersionException) {
+            Map<K, Throwable> unrecoverableFailures =
+                handler.handleUnsupportedVersionException(
+                    (UnsupportedVersionException) t,
+                    spec.keys,
+                    spec.scope instanceof FulfillmentScope);
```

Comment:
If the scope is NOT a FulfillmentScope, I think that we don't have to `completeExceptionally` but rather to `retryLookup` like we do in the previous branch. Am I wrong? If this is true, it may be better to completely separate the two cases. For instance, we could also add `handleUnsupportedVersionException` to the lookup strategy in the future.

Reply:
Yes, I agree with you, we can remove `isFulfillmentStage` it from `handleUnsupportedVersionException`, and add another `handleUnsupportedVersionException` to `AdminApiLookupStrategy` if necessary in the future, currently, we can just `retryLookup`.

Reply:
Yeah, that makes sense.
I initially thought about this but didn't want to expand on it too much because then it might also make sense to compact it with the `NoBatchedFindCoordinatorsException`/`NoBatchedOffsetFetchRequestException` case slightly above, so I was concerned that that many changes unrelated to the ListOffsets API might be undesirable.